restapi.go

```diff
@@ -2933,3 +2933,29 @@ func (s *Session) GuildScheduledEventUsers(guildID, eventID string, limit int, w
 	err = unmarshal(body, &st)
 	return
 }
+
+// SearchGuildMembers returns a list of guild member objects whose username or nickname starts with a provided string
+// guildID  : The ID of a Guild
+// query    : Query string to match username(s) and nickname(s) against
+// limit    : Max number of members to return (1-1000)
+func (s *Session) SearchGuildMembers(guildID, query string, limit int) (st []*Member, err error) {
```

Comment:
I think it would be better to name it `GuildMembersSearch`, to be consistent with the endpoint and other `Session` methods.